### PR TITLE
Added documentation for orchestration platforms

### LINKED ISSUE #2825: docs: document distributed Minio deployment using kubernetes (k8s)

### PR DESCRIPTION
**Pull request by nitisht**
## Description
Added documentation for orchestration platforms

## Motivation and Context
It is required for community members to get started with Minio 
on various platforms like Kubernetes, DC/OS and Swarm
Fixes #2825 https://github.com/minio/minio/issues/3489 https://github.com/minio/cookbook/issues/97

## How Has This Been Tested?
Executed locally. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by harshavardhana** [comment]
Please squash all the commits @NitishT

File: docs/orchestration/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+> In a cloud native environment, scalability is not a function of the application but the orchestration platform.
+
+## Orchestration platforms
+
+Below is the list of documented orchestration platforms for Minio server: 
+
+| Orchestration platforms|
+|:---|
+| [`Kubernetes`](./kubernetes) |
+| [`Docker Swarm`](./docker-swarm) |
```
Comment (harshavardhana) [inline docs/orchestration/README.md]: You should put a RAW url here..

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: docs/orchestration/README.md [hunk 2]
```diff
@@ -3,20 +3,20 @@
...
 
-Below is the list of documented orchestration platforms for Minio server: 
+Below is the list of documented orchestration platforms for Minio server:
 
 | Orchestration platforms|
 |:---|
-| [`Kubernetes`](./kubernetes) |
-| [`Docker Swarm`](./docker-swarm) |
-| [`DC/OS`](./dcos) |
+| [`Kubernetes`](./kubernetes/README.md) |
```
Comment (harshavardhana) [inline docs/orchestration/README.md]: You need to point to full URL not just this like this 

```
[`Kubernetes`](https://raw.githubusercontent.com/minio/minio/master/docs/orchestration/kubernetes/README.md)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: docs/orchestration/README.md (hunk 2 above)
Comment (nitisht) [reply docs/orchestration/README.md]: done

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Pending on @abperiasamy review

**Comment by pascalandy** [comment]
When this is merged, I will add my use case. 
https://github.com/minio/minio/issues/3713#issuecomment-279722533

**Comment by nitisht** [comment]
That will be great @pascalandy. We can add your blog to the further reading/community maintained list of Minio + Swarm setups.

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Can you rebase this @NitishT ?

**Comment by nitisht** [comment]
done @harshavardhana

**Review by harshavardhana — APPROVED** [review]
There are some changes needed but we will take this in and work on top of it..

**Pull request merged**